lib/ui/profitloss/RecordAmount.dart

```diff
@@ -0,0 +1,373 @@
+              : viewModel.actions[_Constants.firstListItem].selectedDate,
+        ),
+        parent: this,
+      ),
+    );
+
+    listBuilder.add(ListDivider.build());
+
+    listBuilder.add(
+      RecordAmountListItem(
+        viewModel: RecordAmountListItemViewModel(
+          type: RecordCellType.pickItem,
+          isEditable: viewModel.isEditable,
+          selectedItem: viewModel.isEditable
+              ? selectedCrop
+              : viewModel.actions[_Constants.secondListItem].selectedItem,
+          listOfCrops: viewModel.actions[_Constants.secondListItem].listOfCrops,
+        ),
+        parent: this,
+      ),
+    );
+
+    listBuilder.add(ListDivider.build());
+
+    listBuilder.add(
+      RecordAmountListItem(
+        viewModel: RecordAmountListItemViewModel(
+          type: RecordCellType.description,
+          isEditable: viewModel.isEditable,
+          description: viewModel.isEditable
+              ? description
+              : viewModel.actions[_Constants.thirdListItem].description,
+        ),
+        style: RecordAmountListItemStyles.biggerStyle,
+        parent: this,
+      ),
+    );
+
+    if (viewModel.isEditable) {
+      listBuilder.add(_buildFooter(viewModel, style));
+    }
+    return listBuilder;
+  }
+
+  Padding _buildFooter(
+      RecordAmountViewModel viewModel, RecordAmountStyle style) {
+    return Padding(
+      padding: style.buttonEdgePadding,
+      child: Row(
+        children: <Widget>[
+          Expanded(
+            child: !widget._viewModel.isFilled
+                ? RoundedButton(
+                    viewModel: RoundedButtonViewModel(
+                        title: viewModel.buttonTitle, onTap: null),
+                    style:
+                        RoundedButtonStyle.largeRoundedButtonStyle().copyWith(
+                      backgroundColor: Color(0xFFe9eaf2),
+                    ),
+                  )
+                : RoundedButton(
+                    viewModel: RoundedButtonViewModel(
+                        title: viewModel.buttonTitle, onTap: saveData),
+                    style: viewModel.type == RecordType.sale
+                        ? RoundedButtonStyle.largeRoundedButtonStyle().copyWith(
+                            backgroundColor: Color(0xFF24d900),
+                          )
+                        : RoundedButtonStyle.largeRoundedButtonStyle().copyWith(
+                            backgroundColor: Color(0xFFff8d4f),
+                          ),
+                  ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  checkIfFilled() {
+    setState(() {
+      if (amountIsFilled && cropIsFilled) {
+        widget._viewModel.isFilled = true;
+      } else {
+        widget._viewModel.isFilled = false;
+      }
+    });
+  }
+
+  saveData() {
+    RecordData save = RecordData(
+      amount: amount,
+      date: selectedDate,
+      crop: selectedCrop,
+      description: description,
+    );
+
+    //TODO: Temporal print
```

Comment:
Add a callback function on the `ViewModel` returning the `RecordData` values. Also on the playground pass the function and show a `SnackBar` or a simple popup message saying: 

> Returned: cost: XX.XX // Date: (date selected) // Crop: (Crop selected) // Description: (description)

To test if it's working fine. Then if we have to use it we shouldn't worry of it anymore.

We should merge to develop when it's fully working